pypsa/statistics.py

```diff
@@ -220,7 +220,7 @@ def filter_active_assets(n, c, df: [pd.Series, pd.DataFrame]):
     if not isinstance(n.snapshots, pd.MultiIndex) or isinstance(df, pd.DataFrame):
         return df
     per_period = {}
-    for p in n.snapshots.get_level_values(0):
+    for p in n.snapshots.get_level_values(0).drop_duplicates():
```

Comment:
```suggestion
    for p in n.snapshots.unique(0):
```
even a bit more efficient